### ISSUE TITLE
My short program following the user manual isn't working

### ISSUE THREAD

**Issue by Rorybabory** [DanielChappuis/reactphysics3d#184]
Sorry if this isn't the correct place to ask this question.
[PhysicsSystem.txt](https://github.com/DanielChappuis/reactphysics3d/files/5833081/PhysicsSystem.txt)

**Comment by DanielChappuis** [comment]
Hello.

Could you please be a bit more specific? What isn't working? Do you get an error somewhere? Which line?

**Comment by Rorybabory** [comment]
Sorry I should have been more specific. I have a box and a ground set up, and the box always passes right through the ground.

**Comment by DanielChappuis** [comment]
There must be something wrong somewhere but I can't figure out what exactly (I haven't tested your code). Can you confirm that when you call your Update(double FPS) method you always use the same FPS value for each call ?

**Comment by Rorybabory** [comment]
I tried with both a constant frame rate and a variable frame rate

**Comment by DanielChappuis** [comment]
I have found the issue. You call the following method to make the ground rigid body static.

`body->setType(BodyType::STATIC);`

This sets the inverse mass of the body to be zero. However, after that you call the following method to compute the mass and inertia tensor of your rigid body:

`body->updateMassPropertiesFromColliders();`

With this call, the inverse mass of the body is recomputed and will overwrite the previous one (that was zero).

Therefore, the ground body is not really static anymore and is also falling down. That's why there was no collision between the two bodies.

I will update the code to make sure that the updateMassPropertiesFromColliders() method won't overwrite the inverse mass if the body is static but for the moment, you can simply make sure to call the setType() method after the updateMassPropertiesFromColliders() method. Note that you do not  really need to call this method for a static body.

**Comment by Rorybabory** [comment]
Thanks so much for the help! The explanation made quite a bit of sense, however what is strange is that I added that function in after I was having it fall through the floor, but now that I removed it, it no longer falls through the floor. It likely seems like before I submitted the issue, there were multiple causes, and when I submitted it, that was the only cause. Again, thanks so much for the help!
